examples/cms/src/data.ts

```diff
@@ -0,0 +1,150 @@
+import { CollectionItemData, ManagedCollection, ManagedCollectionField, framer } from "framer-plugin"
+import { PLUGIN_KEYS } from "./constants"
+
+export type ManagedCollectionFieldType = ManagedCollectionField["type"]
+
+type DataSourceField =
+    | {
+          type: ManagedCollectionFieldType
+      }
+    | {
+          type: "html"
+      }
+
+export interface DataSource {
+    id: string
+    fields: Record<string, DataSourceField>
+    items: Record<string, unknown>[]
+}
+
+export function getDataSourcesIds(): string[] {
+    return ["articles", "categories"]
+}
+
+export async function getDataSource(dataSourceId: string): Promise<DataSource> {
+    return await fetch(`/datasources/${dataSourceId}.json`).then(response => response.json())
+}
+
+export function computeFieldsFromDataSource(dataSource: DataSource): ManagedCollectionField[] {
+    const fields: ManagedCollectionField[] = []
+
+    for (const [fieldId, field] of Object.entries(dataSource.fields)) {
+        let type = field.type
+
+        /**
+         * Here you can add support for other types, usually you would want to convert your APIs types to Framer types.
+         */
+
+        if (type === "html") {
+            type = "formattedText"
+        }
+
+        fields.push({
+            id: fieldId,
+            name: fieldId,
+            type,
+            userEditable: false,
+        } as ManagedCollectionField)
+    }
```

Comment:
A cast like `as ManagedCollectionField` is bad practice. Because you lose some type safety. I think it's better to have more repetition than a cast. I think you can prevent doing the cast using a switch statement on the field type